zipline/utils/cache.py

```diff
@@ -334,7 +353,11 @@ def mkdir(self, *path_parts):
             The parts of the path after the working directory.
         """
         path = self.getpath(*path_parts)
-        os.mkdir(path)
+        try:
```

Comment:
If `ensure_dir` makes subdirectories when the parent doesn't exist, then that change will work. The reason for using `makedirs` as opposed to `mkdir` is that `mkdir` fails when the parents don't exist.